### PR TITLE
[rhosp16] first version

### PR DESCRIPTION
**Pull request by yong-huang**

### CODE REVIEW

File: cinder/README.md [hunk 1]
```diff
@@ -27,20 +27,22 @@ The formal Dell EMC container image is published to [Red Hat Container Catalog](
...
 Then, tag and push it to the local registry.
 
 ```bash
-$ docker tag registry.connect.redhat.com/dellemc/openstack-cinder-volume-dellemc 192.168.139.1:8787/dellemc/openstack-cinder-volume-dellemc
-$ docker push 192.168.139.1:8787/dellemc/openstack-cinder-volume-dellemc
+$ docker tag registry.connect.redhat.com/dellemc/openstack-cinder-volume-dellemc-rhosp16  192.168.139.1:8787/dellemc/openstack-cinder-volume-dellemc-rhosp16
+$ docker push 192.168.139.1:8787/dellemc/openstack-cinder-volume-dellemc-rhosp16
 ```
 
 #### Prepare custom environment yaml
```
Comment (Murray-LIANG) [inline cinder/README.md:48]: Better to add a sequence number here like `#### 2. Prepare custom enviroment yaml` although you already give it a Header style.

File: cinder/README.md [hunk 2]
```diff
@@ -183,7 +186,7 @@ For a full detailed instruction of options, please refer to [Unity backend confi
 
 ```bash
 (undercloud) $ openstack overcloud deploy --templates \
-  -e /home/stack/templates/overcloud_images.yaml \
+  -e /home/stack/templates/containers-prepare-parameter.yaml \
   -e /home/stack/templates/custom-dellemc-container.yaml \
   -e /home/stack/templates/cinder-backend-dellemc-unity.yaml \
   -e <other templates>
```
Comment (Murray-LIANG) [inline cinder/README.md:192]: A general question: does OSP16 support configuring certificates for Cinder backend?

File: cinder/README.md [hunk 3]
```diff
@@ -51,12 +53,13 @@ Create or edit `/home/stack/templates/custom-dellemc-container.yaml`.
 
 ```yaml
```
Comment (Murray-LIANG) [inline cinder/README.md:54]: Does OPS16 Cinder support IPv6?

File: cinder/README.md (hunk 1 above)
Comment (yong-huang) [reply cinder/README.md]: Good suggestion. Updated.

File: cinder/README.md (hunk 2 above)
Comment (yong-huang) [reply cinder/README.md:192]: I remember no such configuration for Cinder Unity/VNX, right?

File: cinder/README.md (hunk 3 above)
Comment (yong-huang) [reply cinder/README.md:54]: Yes, I think so.

File: cinder/README.md [hunk 4]
```diff
@@ -16,49 +16,52 @@ The Dell EMC Cinder container image contains following RPM packages:
...
+#### 1. Prepare Dell EMC container
 
 The formal Dell EMC container image is published to [Red Hat Container Catalog](https://access.redhat.com/containers/)
 
 Red Hat OpenStack Platform supports remote registry and local registry for overcloud deployment. In this document, we only introduce local registry.
 
-> in below examples, 192.168.139.1:8787 acts as a local registry.
+> In below examples, 192.168.139.1:8787 acts as a local registry.
+
+**Notes:** We will not introduce how to setup local registry with docker-registry or docker-distribution. In RHOSP16,  undercloud also acts as a local registry, you could push Dell EMC container images to it, and pull these images when deploying overcloud, please refer Red Hat documents.
```
Comment (amagicsun) [inline cinder/README.md:32]: please refer to Red Hat documents.
If there is link pointing to the documents, that will be better.

File: manila/README.md [hunk 5]
```diff
@@ -105,49 +111,52 @@ emc_ssl_cert_verify=False
...
-#### Prepare Dell EMC container
+#### 1. Prepare Dell EMC container
 
 The formal Dell EMC container image is published to [Red Hat Container Catalog](https://access.redhat.com/containers/)
 
 Red Hat OpenStack Platform supports remote registry and local registry for overcloud deployment. In this document, we only introduce local registry.
 
 > in below examples, 192.168.139.1:8787 acts as a local registry.
 
+**Notes:** We will not introduce how to setup local registry with docker-registry or docker-distribution. In RHOSP16,  undercloud also acts as a local registry, you could push Dell EMC container images to it, and pull these images when deploying overcloud, please refer Red Hat documents.
```
Comment (amagicsun) [inline manila/README.md:127]: Same comment

File: cinder/README.md (hunk 2 above)
Comment (Murray-LIANG) [reply cinder/README.md:192]: These are configurations for ssl verification and certificate. You could check the codes.

File: cinder/README.md (hunk 3 above)
Comment (Murray-LIANG) [reply cinder/README.md:54]: Why is IPv6 support mentioned in Manila, but not in Cinder? Is there any additional configuration needed for Manila?

File: cinder/README.md (hunk 2 above)
Comment (yong-huang) [reply cinder/README.md:192]: You mean driver_ssl_cert_verify and driver_ssl_cert_path defined in Cinder framework? We didn't add these options, we only added the options defined in our drivers.

File: cinder/README.md (hunk 3 above)
Comment (yong-huang) [reply cinder/README.md:54]: Yes, there is one more configuration need to set in Manila: 'ManilaIPv6: true'. In Cinder, specify san_ip with ipv6 format is okay.

File: cinder/README.md (hunk 4 above)
Comment (yong-huang) [reply cinder/README.md]: Done.

File: manila/README.md (hunk 5 above)
Comment (yong-huang) [reply manila/README.md]: Done.

File: cinder/README.md [hunk 6]
```diff
@@ -16,49 +16,52 @@ The Dell EMC Cinder container image contains following RPM packages:
...
+#### 1. Prepare Dell EMC container
 
 The formal Dell EMC container image is published to [Red Hat Container Catalog](https://access.redhat.com/containers/)
 
 Red Hat OpenStack Platform supports remote registry and local registry for overcloud deployment. In this document, we only introduce local registry.
 
-> in below examples, 192.168.139.1:8787 acts as a local registry.
+> In below examples, 192.168.139.1:8787 acts as a local registry.
+
+**Notes:** We will not introduce how to setup local registry with docker-registry or docker-distribution. In RHOSP16,  undercloud also acts as a local registry, you could push Dell EMC container images to it, and pull these images when deploying overcloud, please refer Red Hat document: [4.11. Undercloud container registry](https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.0/html/director_installation_and_usage/installing-the-undercloud#undercloud-container-registry).
```
Comment (amagicsun) [inline cinder/README.md:32]: refer => refer to

File: manila/README.md [hunk 7]
```diff
@@ -105,49 +111,52 @@ emc_ssl_cert_verify=False
...
-#### Prepare Dell EMC container
+#### 1. Prepare Dell EMC container
 
 The formal Dell EMC container image is published to [Red Hat Container Catalog](https://access.redhat.com/containers/)
 
 Red Hat OpenStack Platform supports remote registry and local registry for overcloud deployment. In this document, we only introduce local registry.
 
 > in below examples, 192.168.139.1:8787 acts as a local registry.
 
+**Notes:** We will not introduce how to setup local registry with docker-registry or docker-distribution. In RHOSP16,  undercloud also acts as a local registry, you could push Dell EMC container images to it, and pull these images when deploying overcloud, please refer Red Hat document: [4.11. Undercloud container registry](https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.0/html/director_installation_and_usage/installing-the-undercloud#undercloud-container-registry).
```
Comment (amagicsun) [inline manila/README.md:127]: Same comment

File: cinder/README.md (hunk 2 above)
Comment (Murray-LIANG) [reply cinder/README.md:192]: Yes. The user can set these two options when he/she wants to enable ssl verify. All the options defined in Cinder framework can be used in the backend. So, I am wondering how does OSP specify the driver_ssl_cert_path in the container. It is a file path. How is this file on the host mounted into the container? Does RH have offical way to do this? Do we have any best practice for this?

File: cinder/README.md (hunk 6 above)
Comment (yong-huang) [reply cinder/README.md]: Done.

File: manila/README.md (hunk 7 above)
Comment (yong-huang) [reply manila/README.md]: Done.

File: cinder/README.md (hunk 2 above)
Comment (yong-huang) [reply cinder/README.md:192]: We only cover very few scenarios in our verification, to answer these questions, we need to spend more time to investigate them, this is another user story, we couldn't resolve them in this document.

Comment (Murray-LIANG) [reply cinder/README.md:192]: OK. Please add it to the backlog.

**Pull request merged**